external-api/src/http/external_match.rs

```diff
@@ -65,28 +66,48 @@ pub struct ExternalOrder {
     pub base_mint: BigUint,
     /// The side of the market this order is on
     pub side: OrderSide,
-    /// The amount of the order
-    pub amount: Amount,
+    /// The base amount of the order
+    #[serde(default, alias = "amount")]
+    pub base_amount: Amount,
+    /// The quote amount of the order
+    #[serde(default)]
+    pub quote_amount: Amount,
```

Comment:
We should explicitly check for one of these fields to exist and return an error if not.

Reply:
Should also error if both exist, since then it's ambiguous what to set as the `base_amount`.

Reply:
done